### PR TITLE
Features enabled by default

### LINKED ISSUE #322: Have all features enabled by default
At the moment only traffic mirroring is enabled by default, and everything else has to be enabled explicitly. Starting with version 3.0.0, we want to have all features (=mirroring, outgoing traffic, file ops, env vars) enabled by default, with the option to disable them explicitly.

### PR DESCRIPTION
**Pull request by DmitryDodzin**
Enable by default most of the reasonable features

closes #322

### CODE REVIEW

**Review by eyalb181 — CHANGES_REQUESTED** [review]
LGTM. However, we can only merge this before version 3.0.0, since it's a major API change. I'm marking this with "Request Changes" for now so we don't merge it accidentally.

File: mirrord-cli/src/config.rs [hunk 1]
```diff
@@ -88,16 +88,16 @@ pub(super) struct ExecArgs {
...
     pub tcp_steal: bool,
 
-    /// Enable tcp outgoing feature.
-    #[clap(short = 'o', long, value_parser)]
-    pub enable_tcp_outgoing: bool,
+    /// Disable tcp outgoing feature.
+    #[clap(long = "no-outgoing", value_parser)]
+    pub no_tcp_outgoing: bool,
 
     /// Enable udp outgoing feature.
```
Comment (aviramha) [inline mirrord-cli/src/config.rs:98]: fix comment (disable)
p.s no need to specify the long param as it's same

File: mirrord-cli/src/config.rs [hunk 2]
```diff
@@ -39,25 +39,25 @@ pub(super) struct ExecArgs {
...
-
-    /// Enable file read only - writes will occur locally
+    /// Disable file hooking (Both R/W)
     #[clap(long, value_parser)]
-    pub enable_ro_fs: bool,
+    pub no_fs: bool,
+
+    /// Force file read only - writes will occur locally
+    #[clap(long = "ro", value_parser)]
+    pub ro_fs: bool,
```
Comment (aviramha) [inline mirrord-cli/src/config.rs:48]: ro_fs should be default

**Review by aviramha — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,10 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 - Socket ops - `connect`: ignore localhost and ports 50000 - 60000 (reserved for debugger)
 - Add "*.plist" to `IGNORE_REGEX`, refer [[#350](https://github.com/metalbear-co/mirrord/issues/350)].
 
+### Changed
+- Change all functionality (incoming traffic mirroring, remote DNS outgoing traffic, environment variables, file reads and writes) to be enabled by default. ***Note that flags now disable functionality***
```
Comment (aviramha) [inline CHANGELOG.md:17]: file reads

File: mirrord-cli/src/config.rs [hunk 4]
```diff
@@ -39,25 +39,25 @@ pub(super) struct ExecArgs {
...
-    /// Enable file hooking (Both R/W)
-    #[clap(short = 'f', long, value_parser)]
-    pub enable_fs: bool,
+    /// Disable file hooking (Both R/W)
+    #[clap(long = "rw", value_parser)]
+    pub enable_rw_fs: bool,
 
-    /// Enable file read only - writes will occur locally
+    /// Force file read only - writes will occur locally
     #[clap(long, value_parser)]
```
Comment (aviramha) [inline mirrord-cli/src/config.rs:47]: maybe switch this to enum? `--file rw/r/off`?

File: mirrord-cli/src/config.rs [hunk 5]
```diff
@@ -39,25 +39,25 @@ pub(super) struct ExecArgs {
...
-    #[clap(short = 'f', long, value_parser)]
-    pub enable_fs: bool,
-
-    /// Enable file read only - writes will occur locally
+    /// Disable file read only
     #[clap(long, value_parser)]
-    pub enable_ro_fs: bool,
+    pub no_fs: bool,
+
+    /// Disable file hooking (Both R/W)
```
Comment (eyalb181) [inline mirrord-cli/src/config.rs:46]: Disable?

File: mirrord-cli/src/config.rs [hunk 6]
```diff
@@ -79,7 +79,7 @@ pub(super) struct ExecArgs {
     #[clap(value_parser)]
     pub(super) binary_args: Vec<String>,
 
-    /// Where to extract the library to (defaults to a temp dir)
+    /// Where to extract the library to. Defaults is temp dir.
```
Comment (eyalb181) [inline mirrord-cli/src/config.rs:82]:
```suggestion
    /// Where to extract the library to. Default is temp dir.
```

File: mirrord-cli/src/main.rs [hunk 7]
```diff
@@ -134,16 +134,16 @@ fn exec(args: &ExecArgs) -> Result<()> {
         std::env::set_var("MIRRORD_AGENT_TTL", agent_ttl.to_string());
     }
 
-    if args.enable_fs && args.enable_ro_fs {
-        warn!("Both filesystem read write and read only enabled - RW will take precedence");
+    if args.enable_rw_fs && args.no_fs {
+        warn!("Both fs flags Read/Write and disabled Writes - Disabling will take precedence");
```
Comment (eyalb181) [inline mirrord-cli/src/main.rs:138]:
```suggestion
        warn!("fs was both enabled and disabled - disabling will take precedence.");
```

File: mirrord-layer/src/file.rs [hunk 8]
```diff
@@ -40,6 +40,7 @@ static IGNORE_FILES: LazyLock<RegexSet> = LazyLock::new(|| {
         r".*\.js",
         r".*\.pth",
         r".*\.plist",
+        r".*venv\.cfg",
```
Comment (eyalb181) [inline mirrord-layer/src/file.rs:43]: Is this related to the PR?

**Review by eyalb181 — CHANGES_REQUESTED** [review]

**Review by aviramha — APPROVED** [review]

**Pull request merged**